packages/flutter_tools/lib/src/commands/multi_drive.dart

```diff
@@ -0,0 +1,375 @@
+import 'dart:async';
+import 'dart:convert' show JSON;
+import 'dart:io' as io;
+
+import 'package:path/path.dart' as path;
+import 'package:test/src/executable.dart' as executable; // ignore: implementation_imports
+
+import '../android/android_device.dart' show AndroidDevice;
+import '../application_package.dart';
+import '../base/file_system.dart';
+import '../base/common.dart';
+import '../base/logger.dart';
+import '../base/os.dart';
+import '../base/process.dart';
+import '../build_info.dart';
+import '../dart/sdk.dart';
+import '../device.dart';
+import '../globals.dart';
+import '../ios/simulators.dart' show SimControl, IOSSimulatorUtils;
+import '../run.dart';
+import '../runner/flutter_command.dart';
+import 'build_apk.dart' as build_apk;
+import 'run.dart';
+
+class MultiDriveCommand extends FlutterCommand {
+  MultiDriveCommand() {
+    addBuildModeFlags(defaultToRelease: false);
+
+    argParser.addFlag('trace-startup',
+        negatable: true,
+        defaultsTo: false,
+        help: 'Start tracing during startup.');
+
+    /*
+    argParser.addOption('route',
+        help: 'Which route to load when running the app.');
+    */
+
+    argParser.addFlag(
+      'keep-app-running',
+      negatable: true,
+      defaultsTo: false,
+      help:
+        'Will keep the Flutter application running when done testing. By '
+        'default Flutter Driver stops the application after tests are finished.'
+    );
+
+    argParser.addFlag(
+      'use-existing-app',
+      negatable: true,
+      defaultsTo: false,
+      help:
+        'Will not start a new Flutter application but connect to an '
+        'already running instance. This will also cause the driver to keep '
+        'the application running after tests are done.'
+    );
+
+    // Add --specs option to load the specs file that specifies test path,
+    // device ids, application paths and debug ports
+    argParser.addOption(
+      'specs',
+      defaultsTo: null,
+      allowMultiple: false,
+      help:
+        'Config file that specifies the devices, apps and debug-ports for testing.'
+    );
+
+    // argParser.addOption(
+    //   'debug-ports',
+    //   defaultsTo: kDefaultMultiDrivePort,
+    //   allowMultiple: true,
+    //   splitCommas: true,
+    //   help: 'Listen to a list of ports for a debug connection.'
+    // );
+  }
+
+  bool get traceStartup => argResults['trace-startup'];
+
+  String get route => null;//argResults['route'];
```

Comment:
The default value for 'trace-startup' in the flutter source code is null, so it's better to conform with the flutter team if we use the route argument.  We may change this later if it turns out that we need to specify this argument to non-null value.  Thanks for pointing this out.